src/adapters/scene.js

```diff
@@ -118,6 +120,11 @@ Scene.prototype.initMapBox = function ({ locationHash, bbox }) {
   this.lastDoubleTapTimeStamp = 0;
   this.lastTouchEndTimeStamp = 0;
   this.mb.on('touchend', _e => {
+    // Start a new 2s idle timeout
+    MOBILE_IDLE_TIMEOUT = setTimeout(() => {
```

Comment:
Instead of having to duplicate that everywhere and separate the timeout creation and clear, a better way would be to listen to the existing `map_user_interaction` event, which is already used for something similar (maybe with more conditions to fire it).
That way, everything would be in the same place, with no duplication and with a straightforward algo: on *any* map interaction, show the scale and start the timeout.

Reply:
The way I see it:

the timer should start when the map has loaded (in scene.js)

It should stop / reset when we interact with the map (touch, pan, click, long click, zoom)
(we could put that in map_user_interaction, which is in PanelManager.jsx)

It should restart when the map is not interacted with anymore (touchend, moveend, ...)

How would you proceed to share the same timer between these two components?
Or would you simply put a simple "stop-reset-restart" in map_user_interaction, and nothing else?

Reply:
What I propose:
- Remove all these `setTimeout` in all these different event
- Factorize them in a single one, in a listener to `map_user_interaction`, which is already thrown on pans and zooms, maybe it needs to be thrown on more map events.
- Put the listener where you want. The one in `PanelManager` deals with the panel minimization, so it's in its right place, but here we want to deal with map controls, it's probably better to define a second listener for it in `scene.js` or somewhere else.
- In the listener, I think it should be enough to clear the timeout each time we receive the event and start a new one

Reply:
I did not fire map_user_interaction on other events than the ones that already do it because map_user_interaction  minimizes the current panel, and for the other events (click, long touch) we do NOT want the current panel to minimize.

So instead, in these events, I'll fire directly my new event: restart_idle_timeout, whose listener is placed in scene.js

I hope it's ok for you!

Reply:
Alright, as long as the timeout code isn't duplicated it's good.